jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -111,11 +113,48 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return false;
         }
         Handler handler = matchedResource.getResource();
+        PathSpec pathSpec = matchedResource.getPathSpec();
         if (LOG.isDebugEnabled())
             LOG.debug("Matched {} to {} -> {}", pathInContext, matchedResource.getPathSpec(), handler);
-        boolean handled = handler.handle(request, response, callback);
+
+        PathSpecRequest pathSpecRequest = new PathSpecRequest(request, pathSpec);
```

Comment:
Does the contextualization of a request make sense for non prefix matches?   Perhaps this is better in a subclass that only allows prefix patterns?